presto-tpcds/src/test/java/com/facebook/presto/tpcds/TestTpcdsRecordSet.java

```diff
@@ -0,0 +1,124 @@
+import static com.teradata.tpcds.column.CallCenterColumn.CC_REC_END_DATE;
+import static com.teradata.tpcds.column.CallCenterColumn.CC_REC_START_DATE;
+import static com.teradata.tpcds.column.CallCenterColumn.CC_SQ_FT;
+import static com.teradata.tpcds.column.CallCenterColumn.CC_STATE;
+import static com.teradata.tpcds.column.CallCenterColumn.CC_TAX_PERCENTAGE;
+import static com.teradata.tpcds.column.DbgenVersionColumn.DV_CREATE_DATE;
+import static com.teradata.tpcds.column.DbgenVersionColumn.DV_CREATE_TIME;
+import static com.teradata.tpcds.column.DbgenVersionColumn.DV_VERSION;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestTpcdsRecordSet
+{
+    private final Session session = Session.getDefaultSession().withScale(.01);
+
+    @Test
+    public void testGetColumnTypes()
+    {
+        Table table = Table.getTable(Table.CALL_CENTER.getName());
+        List<Column> columns = ImmutableList.of(
+                CC_CALL_CENTER_SK,  // Big Int
+                CC_SQ_FT,           // Integer
+                CC_NAME,            // Varchar
+                CC_STATE,           // Character
+                CC_TAX_PERCENTAGE); // Decimal
+        RecordSet recordSet = new TpcdsRecordSet(Results.constructResults(table, session), columns);
+        assertEquals(recordSet.getColumnTypes(), ImmutableList.of(
+                BIGINT,
+                INTEGER,
+                createVarcharType(50),
+                createCharType(2),
+                createDecimalType(5, 2)));
+
+        table = Table.getTable(Table.DBGEN_VERSION.getName());
+        columns = ImmutableList.of(
+                DV_CREATE_DATE, // Date
+                DV_CREATE_TIME, // Time
+                DV_VERSION);      // Varchar
+        recordSet = new TpcdsRecordSet(Results.constructResults(table, session), columns);
+        assertEquals(recordSet.getColumnTypes(), ImmutableList.of(
+                DATE,
+                TIME,
+                createVarcharType(16)));
+
+        table = Table.getTable(Table.DBGEN_VERSION.getName());
+        columns = ImmutableList.of();
+        recordSet = new TpcdsRecordSet(Results.constructResults(table, session), columns);
+        assertEquals(recordSet.getColumnTypes(), ImmutableList.of());
+    }
+
+    @Test
+    public void testCursor()
+    {
+        Table table = Table.getTable(Table.CALL_CENTER.getName());
+        Results result = Results.constructResults(table, session);
+        RecordSet recordSet = new TpcdsRecordSet(result, ImmutableList.copyOf(table.getColumns()));
+
+        try (RecordCursor cursor = recordSet.cursor()) {
+            if (cursor.advanceNextPosition()) {
+                // Date
+                assertFalse(cursor.isNull(CC_REC_START_DATE.ordinal()));
+                assertEquals(cursor.getLong(CC_REC_START_DATE.ordinal()), 10227);
```

Comment:
I'm noticing slight differences with epoch date values with joda and java time. Seems like this has been a thing since [2014](https://github.com/JodaOrg/joda-time/issues/190#issuecomment-71255053) despite docs stating [1970-01-01](https://www.joda.org/joda-time/apidocs/org/joda/time/LocalDate.html#LocalDate-long-)
![Screen Shot 2022-03-28 at 5 23 50 PM](https://user-images.githubusercontent.com/16763902/160497985-0429e845-658e-4729-9562-fa4a5109f647.png)

Technically the test should assert for 10,228 days since joda-epoch, but I'm opting 10,227 since JodaOrg is asking users "to migrate to java.time (JSR-310) - a core part of the JDK which replaces this project."

Reply:
Min select on the call center table yields 1998-01-01 and it's represented as a [string](https://github.com/Teradata/tpcds/blob/c0541d36946091072f02c3a1eaaafa1c57f50e4b/src/main/java/com/teradata/tpcds/row/CallCenterRow.java#L246).
![Screen Shot 2022-03-29 at 9 34 43 AM](https://user-images.githubusercontent.com/16763902/160641501-8d44363b-8e67-4d15-9d08-7e8872f70a08.png)


I see a different min value for `web_site.web_rec_start_date_sk`, which might be because it's not using the method to convert the field into a Julian Date string. 
![Screen Shot 2022-03-29 at 10 03 31 AM](https://user-images.githubusercontent.com/16763902/160653829-91f2f230-70eb-450c-af48-35d5bfda7af9.png)

* WebSite
https://github.com/Teradata/tpcds/blob/c0541d36946091072f02c3a1eaaafa1c57f50e4b/src/main/java/com/teradata/tpcds/row/WebSiteRow.java#L184

*Call Center
https://github.com/Teradata/tpcds/blob/c0541d36946091072f02c3a1eaaafa1c57f50e4b/src/main/java/com/teradata/tpcds/row/CallCenterRow.java#L246

I'd like to note that the same behavior is occurring with Joda time, but the dates are off by 1 day--1998-01-01 in java time vs 1998-01-02 in joda time.

Reply:
Thanks for the investigation!  Sounds like java time is more correct.  The start date discrepancy is expected, as they are different for different tables https://github.com/Teradata/tpcds/blob/c0541d36946091072f02c3a1eaaafa1c57f50e4b/src/main/java/com/teradata/tpcds/SlowlyChangingDimensionUtils.java#L42.  Call center is the first table in the schema, so its start date is 1998-01-01.

Reply:
Gotcha, thanks for clarifying that. I wasn't sure if all fields used the minimum date listed [here](https://github.com/Teradata/tpcds/blob/c0541d36946091072f02c3a1eaaafa1c57f50e4b/src/main/java/com/teradata/tpcds/type/Date.java#L35)